calendar/store/welcome_store_test.go

```diff
@@ -0,0 +1,141 @@
+package store
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/mattermost/mattermost/server/public/model"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/testutil"
+)
+
+func TestLoadUserWelcomePost(t *testing.T) {
+	mockAPI, store, _, _, _ := GetMockSetup(t)
+
+	tests := []struct {
+		name       string
+		setup      func(*testutil.MockPluginAPI)
+		assertions func(*testing.T, string, error)
+	}{
+		{
+			name: "Error loading user welcome post",
+			setup: func(mockAPI *testutil.MockPluginAPI) {
+				mockAPI.On("KVGet", MockString).Return(nil, &model.AppError{Message: "KVGet failed"})
+			},
+			assertions: func(t *testing.T, resp string, err error) {
+				require.Equal(t, resp, "")
+				require.EqualError(t, err, "failed plugin KVGet: KVGet failed")
+			},
+		},
+		{
+			name: "Success loading user welcome post",
+			setup: func(mockAPI *testutil.MockPluginAPI) {
+				mockAPI.On("KVGet", MockString).Return([]byte(`"mockPostID"`), nil)
+			},
+			assertions: func(t *testing.T, resp string, err error) {
+				require.NoError(t, err)
+				require.Equal(t, "mockPostID", resp)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockAPI.ExpectedCalls = nil
+			tt.setup(mockAPI)
+
+			resp, err := store.LoadUserWelcomePost("mockMMUserID")
+
+			tt.assertions(t, resp, err)
+
+			mockAPI.AssertExpectations(t)
+		})
+	}
+}
+
+func TestStoreUserWelcomePost(t *testing.T) {
+	mockAPI, store, _, _, _ := GetMockSetup(t)
+
+	tests := []struct {
+		name       string
+		setup      func(*testutil.MockPluginAPI)
+		assertions func(*testing.T, error)
+	}{
+		{
+			name: "Error storing user welcome post",
+			setup: func(mockAPI *testutil.MockPluginAPI) {
+				mockAPI.On("KVSet", MockString, mock.Anything).Return(&model.AppError{Message: "KVSet failed"})
```

Comment:
Let's completely remove all 'mock.Anything' and only use typesafe args.

Reply:
@wiggin77 
Value of type []byte were not allowed to be mocked using `mock.AnythingOfType` so used `mock.MatchedBy` to keep the testcases typesafe
```
mock.MatchedBy(func(arg interface{}) bool {
        _, ok := arg.([]byte)
        return ok
    }),
```